test/e2e/ownerrefs_finalizers_test.go

```diff
@@ -323,26 +392,21 @@ func checkClusterIdentitySecretOwnerRefAndFinalizer(ctx context.Context, c ctrlc
 
 // forcePeriodicReconcile forces the vSphereDeploymentZone and ClusterResourceSets to reconcile every 20 seconds.
 // This reduces the chance of race conditions resulting in flakes in the test.
-func forcePeriodicReconcile(ctx context.Context, c ctrlclient.Client, namespace string) {
+func forcePeriodicReconcile(ctx context.Context, c ctrlclient.Client) {
 	deploymentZoneList := &infrav1.VSphereDeploymentZoneList{}
-	crsList := &addonsv1.ClusterResourceSetList{}
 	ticker := time.NewTicker(20 * time.Second)
 	stopTimer := time.NewTimer(5 * time.Minute)
 	go func() {
 		defer GinkgoRecover()
 		for {
 			select {
 			case <-ticker.C:
-				Expect(c.List(ctx, deploymentZoneList)).To(Succeed())
-				for _, zone := range deploymentZoneList.Items {
-					annotationPatch := ctrlclient.RawPatch(types.MergePatchType, []byte(fmt.Sprintf("{\"metadata\":{\"annotations\":{\"cluster.x-k8s.io/modifiedAt\":\"%v\"}}}", time.Now().Format(time.RFC3339))))
-					Expect(c.Patch(ctx, zone.DeepCopy(), annotationPatch)).To(Succeed())
-				}
-				// TODO: check if this can be dropped after https://github.com/kubernetes-sigs/cluster-api/pull/10656 is merged
-				Expect(c.List(ctx, crsList, ctrlclient.InNamespace(namespace))).To(Succeed())
-				for _, crs := range crsList.Items {
-					annotationPatch := ctrlclient.RawPatch(types.MergePatchType, []byte(fmt.Sprintf("{\"metadata\":{\"annotations\":{\"cluster.x-k8s.io/modifiedAt\":\"%v\"}}}", time.Now().Format(time.RFC3339))))
-					Expect(c.Patch(ctx, crs.DeepCopy(), annotationPatch)).To(Succeed())
+				if testMode == GovmomiTestMode {
```

Comment:
I would prefer not running this func in supervisor mode (see comment above). If we go with that change we could rename the func e.g. to forcePeriodicDeploymentZoneReconcile and then drop this if inside the func